src/mpyl/utilities/docker/__init__.py

```diff
@@ -190,6 +198,10 @@ def push_to_registry(
 
     login(logger=logger, registry_config=docker_config)
     full_image_path = docker_registry_path(docker_config, image_name)
+    repo_path = ecr_repository_path(docker_config.host_name, image_name)
+
+    if docker_config.provider == "aws":
```

Comment:
Oh I was thinking, maybe we can at least type the provider as a literal string? https://mypy.readthedocs.io/en/stable/literal_types.html#parameterizing-literals

Reply:
Why isn't it an enum?

Reply:
Also fine, the are on the same page. The frontend guild is moving from enum to string literals, not sure why precisely

Reply:
added enum class Provider